File: euler/02.c
#include <stdio.h>

int main()
{
	int a = 1, b = 2, c = 0, sum = 2;
	do {
		c = a + b;
		a = b;
		b = c;
		if ( c % 2 == 0) {
			sum += c;
		}
	} while ( b < 4000000 );
	printf("%d\n", sum);
	return 0;
}

File: euler/02.rb
#!/usr/bin/env ruby

a = 1
b = 2
c = 0
sum = 2
while c < 4000000 do
  c = a + b
  a = b
  b = c
  if c % 2 == 0
    sum += c
  end
end
p sum
